tt_metal/hw/ckernels/grayskull/common/inc/ckernel_sfpu_binop_with_unary.h

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel_defs.h"
+#include "ckernel.h"
+#include "noc_nonblocking_api.h"
+#include "ckernel_sfpu_converter.h"
+#include <limits>
+
+#include "sfpi.h"
+
+using namespace sfpi;
+
+namespace ckernel
+{
+namespace sfpu
+{
+
+enum {ADD =0, SUB=1, MUL=2, DIV=3, RSUB=4,};
+
+
+
+template <bool APPROXIMATION_MODE, int BINOP_MODE, int ITERATIONS=8>
+void calculate_binop_with_scalar(uint32_t param)
+{
+
+    const vFloat parameter = Converter::to_float(param);
+    vFloat inv_parameter;
+
+    if  constexpr(BINOP_MODE == DIV)  {
+       inv_parameter = (1.0f/Converter::to_float(param));
+    }
```

Comment:
Host should pass in inv_parameter value instead of computing it in kernel runtime for better perf